pennylane/data/attributes/array.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains AttributeType definition for numpy arrays."""
+
+import numpy
+from numpy.typing import ArrayLike
```

Comment:
implemented!

Reply:
There are tests (see `tests/data/attributes/test_array.py`) that check that `DatasetArray` returns the same interface. The type annotations are mainly informative, and should still be `ArrayLike` since it's the most general - `TensorLike` being a subset of `ArrayLike`

Reply:
other way around - `ArrayLike` is a subset of `TensorLike`. `TensorLike` supports various interfaces. I'm not sure if `pennylane.numpy` arrays are captured by `ArrayLike`, but I'm quite sure that other interfaces aren't so we should probably switch it in anticipation for support for other interfaces.

Reply:
done!